src/lib/workspace/WorkspaceRouter.tsx

```diff
@@ -62,6 +63,76 @@ export function WorkspaceRouter({
   const analysisClient = useConfiguredAnalysisClient(userServiceUrl);
   const downloadClient = useConfiguredDownloadClient(downloadServiceUrl);
 
+  // The following useEffect handles when the user presses the back button and
+  // is inadvertently moved back to a new analysis URL from their saved analysis URL
+  const history = useHistory();
+  // Used to determine whether a POP action was the back or forward button
+  // (it can be either)
+  const [locationKeys, setLocationKeys] = useState<string[]>([]);
+  // A full history stack
+  const pathnamesRef = useRef([history.location.pathname]);
+  // Tracks the current position in the history stack
+  const pathnamesCursorRef = useRef(0);
+
+  useEffect(() => {
+    return history.listen((location) => {
+      if (location.key === undefined) return;
+
+      if (history.action === 'PUSH') {
+        setLocationKeys([location.key]);
+        pathnamesRef.current = pathnamesRef.current
+          .slice(0, pathnamesCursorRef.current + 1)
+          .concat(location.pathname);
+        pathnamesCursorRef.current++;
+      } else if (history.action === 'POP') {
+        // The user has navigated using either the forward or back button
+
+        // Sanity check
+        if (!(locationKeys.length > 0)) return;
+
+        if (locationKeys[1] === location.key) {
+          // This is a forward button press (probably)
+          setLocationKeys(([_, ...keys]) => keys);
+
+          // Sanity check
+          if (!(pathnamesCursorRef.current + 1 < pathnamesRef.current.length))
+            return;
+
+          pathnamesCursorRef.current++;
+        } else {
+          // This is a back button press (probably)
+          setLocationKeys((keys) => [location.key!, ...keys]);
+
+          // Sanity check
+          if (!(pathnamesCursorRef.current > 0)) return;
+
+          pathnamesCursorRef.current--;
+
+          const lastPathname =
+            pathnamesRef.current[pathnamesCursorRef.current + 1];
+          const newAnalysisRegex = /eda\/.*\/new\/.*/;
+          const savedAnalysisRegex = /eda\/[^/]*\/(?!new\/)([^/]*)\/.*/;
```

Comment:
The base url for the EDA workspace isn't necessarily `/eda` (for example, the ClinEpi website uses `/workspace/analyses`), so these regexes will need to be dynamic based on the base url.

Reply:
Good catch! :detective:

Reply:
@jtlong3rd Thanks for the info! How does it look now?

Reply:
The regexes look great! Just to be safe, could you also add `path` to the effect dependencies?